tests/unit/conftest.py

```diff
@@ -1,86 +1,15 @@
-import functools
-import json
-import os
 import re
 from unittest import mock
 
 import httpretty
-import jwt
 import pytest
-import sqlalchemy
 from pyramid import testing
 from pyramid.request import apply_request_extensions
-from sqlalchemy.orm import sessionmaker
 
-from lms import db
-from lms.models import ApplicationInstance
 from lms.services.application_instance_getter import ApplicationInstanceGetter
 from lms.services.launch_verifier import LaunchVerifier
 from lms.values import LTIUser
-
-TEST_DATABASE_URL = os.environ.get(
-    "TEST_DATABASE_URL", "postgresql://postgres@localhost:5433/lms_test"
-)
-
-
-SESSION = sessionmaker()
-
-
-@pytest.fixture(scope="session")
-def db_engine():
-    engine = sqlalchemy.create_engine(TEST_DATABASE_URL)
-    db.init(engine)
-    return engine
-
-
-@pytest.yield_fixture
-def db_session(db_engine):
-    """
-    Yield the SQLAlchemy session object.
-
-    We enable fast repeatable database tests by setting up the database only
-    once per session (see :func:`db_engine`) and then wrapping each test
-    function in a transaction that is rolled back.
-
-    Additionally, we set a SAVEPOINT before entering the test, and if we
-    detect that the test has committed (i.e. released the savepoint) we
-    immediately open another. This has the effect of preventing test code from
-    committing the outer transaction.
-
-    """
-    conn = db_engine.connect()
-    trans = conn.begin()
-    session = SESSION(bind=conn)
-    session.begin_nested()
-
-    @sqlalchemy.event.listens_for(session, "after_transaction_end")
-    def restart_savepoint(session, transaction):  # pylint:disable=unused-variable
-        if (
-            transaction.nested and not transaction._parent.nested
-        ):  # pylint:disable=protected-access
-            session.begin_nested()
-
-    try:
-        yield session
-    finally:
-        session.close()
-        trans.rollback()
-        conn.close()
-
-
-def autopatcher(request, target, **kwargs):
-    """Patch and cleanup automatically. Wraps :py:func:`mock.patch`."""
-    options = {"autospec": True}
-    options.update(kwargs)
-    patcher = mock.patch(target, **options)
-    obj = patcher.start()
-    request.addfinalizer(patcher.stop)
-    return obj
-
-
-@pytest.fixture
-def patch(request):
-    return functools.partial(autopatcher, request)
+from tests.conftests import *
```

Comment:
Is this necessary or does pytest find the higher up `conftest.py` itself anyway?

Reply:
It would if you ran it from that directory, but if you run it from inside `tests/unit` or `tests/functional` as we do, then `pytest` won't recurse outside of the CWD.